save-backend/src/main/kotlin/org/cqfn/save/backend/repository/BaseEntityRepository.kt

```diff
@@ -1,11 +1,20 @@
 package org.cqfn.save.backend.repository
 
 import org.cqfn.save.entities.BaseEntity
+import org.springframework.data.jpa.domain.Specification
 import org.springframework.data.jpa.repository.JpaRepository
 import org.springframework.data.repository.NoRepositoryBean
+import kotlin.reflect.KClass
 
 /**
  * JPA repositories for the subclasses of the BaseEntity
  */
 @NoRepositoryBean
-interface BaseEntityRepository<T : BaseEntity> : JpaRepository<T, Long>
+interface BaseEntityRepository<T : BaseEntity> : JpaRepository<T, Long> {
+    /**
+     * @param entityClass
+     * @param specification
+     * @return list of class objects
+     */
+    fun getList(entityClass: KClass<T>, specification: Specification<T>): List<T>
```

Comment:
I believe you can extend `JpaSpecificationExecutor<T>` to be able to configure query using `Specification`. Did you take a look at [this section of docs](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#specifications)?

Reply:
Yes, I think we can use it :)